config/main.py

```diff
@@ -1410,38 +1437,113 @@ def qos(ctx):
     pass
 
 @qos.command('clear')
-def clear():
+@click.option(
+    '-n', '--namespace', default=None, type=str, help="Network namespace"
+)
+def clear(namespace):
     """Clear QoS configuration"""
     log_info("'qos clear' executing...")
-    _clear_qos()
+    _clear_qos(namespace)
 
 @qos.command('reload')
-def reload():
+@click.option(
+    '-n', '--namespace', default=None, type=str, help="Network namespace"
+)
+def reload(namespace):
     """Reload QoS configuration"""
     log_info("'qos reload' executing...")
-    _clear_qos()
+    _clear_qos(namespace)
     platform = sonic_device_util.get_platform()
     hwsku = sonic_device_util.get_hwsku()
-    buffer_template_file = os.path.join('/usr/share/sonic/device/', platform, hwsku, 'buffers.json.j2')
-    if os.path.isfile(buffer_template_file):
-        command = "{} -d -t {} >/tmp/buffers.json".format(SONIC_CFGGEN_PATH, buffer_template_file)
-        run_command(command, display_cmd=True)
-
-        qos_template_file = os.path.join('/usr/share/sonic/device/', platform, hwsku, 'qos.json.j2')
-        sonic_version_file = os.path.join('/etc/sonic/', 'sonic_version.yml')
-        if os.path.isfile(qos_template_file):
-            command = "{} -d -t {} -y {} >/tmp/qos.json".format(SONIC_CFGGEN_PATH, qos_template_file, sonic_version_file)
-            run_command(command, display_cmd=True)
+    num_npus = sonic_device_util.get_num_npus()
+    namespace_list = [DEFAULT_NAMESPACE]
+    if num_npus > 1:
+        namespace_list = sonic_device_util.get_namespaces()
+
+    if namespace:
+        if namespace not in namespace_list:
+            click.secho(
+                "Command {} failed with invalid namespace {}".format(
+                    "qos reload", namespace
+                ),
+                fg='yellow'
+            )
+            raise click.Abort()
+        namespace_list = [namespace]
 
-            # Apply the configurations only when both buffer and qos configuration files are presented
-            command = "{} -j /tmp/buffers.json --write-to-db".format(SONIC_CFGGEN_PATH)
-            run_command(command, display_cmd=True)
-            command = "{} -j /tmp/qos.json --write-to-db".format(SONIC_CFGGEN_PATH)
+    for ns in namespace_list:
+        if ns is DEFAULT_NAMESPACE:
+            asic_id_suffix = ""
+        else:
+            match = re.match(NETWORK_NAMESPACE_MATCH, ns, re.IGNORECASE)
+            if match:
```

Comment:
There is function to do this [get_npu_id_from_name](https://github.com/Azure/sonic-buildimage/blob/81a9d5ab147576b2d5ecfe790de9e42fc2b8f6e0/src/sonic-config-engine/sonic_device_util.py#L41)